lib/logstash/inputs/gelf.rb

```diff
@@ -96,7 +96,12 @@ def udp_listener(output_queue)
       # Gelfd parser outputs null if it received and cached a non-final chunk
       next if data.nil?
 
-      event = LogStash::Event.new(LogStash::Json.load(data))
+      begin
+        event = LogStash::Event.new(LogStash::Json.load(data))
+      rescue => ex
+        @logger.warn("JSON failed to parse a message. Skipping", :exception => ex, :backtrace => ex.backtrace)
```

Comment:
Here's an example of handling JSON failures in the json codec: https://github.com/logstash-plugins/logstash-codec-json/blob/master/lib/logstash/codecs/json.rb#L52-L54